librz/core/windows_heap.c

```diff
@@ -1337,94 +1328,83 @@ static void w32_list_heaps_blocks(RzCore *core, const char format) {
 			RZ_FREE(heapInfo->heaps[i].Blocks);
 		}
 	}
-	if (format == 'j') {
+	if (mode == RZ_OUTPUT_MODE_JSON) {
 		pj_end(pj);
 		rz_cons_println(pj_string(pj));
-	} else if (format != 'f') {
+	} else if (!flag) {
 		rz_cons_println(rz_table_tostring(tbl));
 	}
 	rz_table_free(tbl);
 	pj_free(pj);
 	RtlDestroyQueryDebugBuffer(db);
 }
 
-static const char *help_msg[] = {
-	"Usage:", " dmh[?|b][f|j]", " # Windows heap parsing commands",
-	"dmw[j]", "", "List process heaps",
-	"dmwb[?] [addr]", "", "List process heap blocks",
-	NULL
-};
-
-static const char *help_msg_block[] = {
-	"Usage:", " dmhb[f|j]", " # Windows heap parsing commands",
-	"dmwb [addr]", "", "List allocated heap blocks",
-	"dmwbf", "", "Create flags for each allocated block",
-	"dmwbj [addr]", "", "Print output in JSON format",
-	NULL
-};
-
-static void cmd_debug_map_heap_block_win(RzCore *core, const char *input) {
-	char *space = strchr(input, ' ');
+//static const char *help_msg[] = {
+//	"Usage:", " dmh[?|b][f|j]", " # Windows heap parsing commands",
+//	"dmw[j]", "", "List process heaps",
+//	"dmwb[?] [addr]", "", "List process heap blocks",
+//	NULL
+//};
+//
+//static const char *help_msg_block[] = {
+//	"Usage:", " dmhb[f|j]", " # Windows heap parsing commands",
+//	"dmwb [addr]", "", "List allocated heap blocks",
+//	"dmwbf", "", "Create flags for each allocated block",
+//	"dmwbj [addr]", "", "Print output in JSON format",
+//	NULL
+//};
+
+static void cmd_debug_map_heap_block_win(RzCore *core, const char *addr, RzOutputMode mode, bool flag) {
+	init_func();
```

Comment:
What is this?

Reply:
`init func` call was present in `cmd_debug_map_heap_win` function (line 1400 in old file)

Reply:
It's a function that gets the pointers to the API. I suggest to rename it into something like `initialize_windows_ntdll_query_api_functions()`.

Reply:
agree. that needs to be renamed. i thought this was something malicious.